app/models/api/user_sessions.rb

```diff
@@ -0,0 +1,80 @@
+require 'dry-validation'
+require 'dry-struct'
+
+module Api::UserSessions
+  module Types
+    include Dry::Types.module
+  end
+
+  StreamSchema = Dry::Validation.Schema do
+    required(:deleted).filled(:bool?)
+    required(:average_value).filled(:float?)
+    required(:measurement_type).filled(:str?)
+  end
+
+  Schema = Dry::Validation.Schema do
+    configure do
+      def self.messages
+        super.merge(en: { errors: { valid_streams: 'streams is not valid' } })
+      end
+    end
+
+    required(:data).each do
+      schema do
+        required(:uuid).filled(:str?)
+        required(:deleted).filled(:bool?)
+        required(:tag_list)
+        required(:title)
+        required(:calibration).filled(:int?)
+        required(:contribute).filled(:bool?)
+        required(:drawable).filled(:int?)
+        required(:start_time).filled(:str?)
+        required(:end_time).filled(:str?)
+        required(:is_indoor).filled(:bool?)
+        required(:latitude).filled(:float?)
+        required(:longitude).filled(:float?)
+        required(:type).filled(:str?)
+        required(:notes).each do
+          schema do
+            required(:numebr).filled(:int?)
+            required(:latitude).filled(:int?)
+            required(:longitude).filled(:int?)
+            required(:date).filled(:str?)
+            required(:text).filled(:str?)
+          end
+        end
+        validate(valid_streams: :streams) do |stream|
+          stream.all? do |sensor_name, stream_data|
+            StreamSchema.call(stream_data).success?
+          end
```

Comment:
that's the only way I discovered to validate hash values without knowing the keys (here keys can be anything, cause they are sensor_names).
Apparently, this feature is coming in the 1.0 version of dry-validation: 
https://github.com/dry-rb/dry-schema/issues/37

Reply:
Good job!